src/power_grid_model/validation/utils.py

```diff
@@ -96,15 +97,15 @@ def update_input_data(input_data: SingleDataset, update_data: SingleDataset):
     return merged_data
 
 
-def update_component_data(component: str, input_data: np.ndarray, update_data: np.ndarray) -> None:
+def update_component_data(component: ComponentType, input_data: np.ndarray, update_data: np.ndarray) -> None:
     """
     Update the data in a numpy array, with another numpy array,
     indexed on the "id" field and only non-NaN values are overwritten.
     """
     for field in update_data.dtype.names:
         if field == "id":
             continue
-        nan = nan_type(component, field, "update")
+        nan = nan_type(component, field, DataType.input)
```

Comment:
i think this is incorrect?
```suggestion
       nan = nan_type(component, field, DataType.update)
```